_posts/2018/2018-02-18-avoid-writing-sql-when-using-activerelation.md

```diff
@@ -0,0 +1,74 @@
+---
+title: 'Avoid Writing SQL When Using ActiveRelation'
+description: 'It's fine, but prefer not to if you can.'
+layout: article
+category: ruby
+image:
+  base: '2018/avoid-writing-sql-when-using-activerelation'
+  alt: 'Coding SQL'
+  source: 'https://unsplash.com/photos/fPkvU7RDmCo'
+  credit: 'Caspar Rubin'
+---
+
+ActiveRelation, which powers the searching and querying engine of ActiveRecord, is a powerful and flexible tool.
+
+
+## Instead of...
+
+...writing literal SQL strings inside an ActiveRelation `#where` method.
+
+```ruby
+Person.where("name = #{ params[:name] } and hidden_at is null")
```

Comment:
Do you want the caps here to show it's SQL...? Just to make it easier for the reader to spot SQL-like text?

Reply:
I'm actually trying to show that the non-caps come through. Although I'm not directly calling it out.

Reply:
I figure that...but why don't you write it how it's expected to be seen...and then you can add an aside around like: by the way, people think you need caps for SQL, when you don't...

..you know..in your funny Andy way.

Then, people instantly recognise: oh, this is SQL.

Reply:
I do use the example lower down with the non-capitalised version. So I'm making the same point.